internal/app/handlers.go

```diff
@@ -460,13 +461,86 @@ func (app *Application) home(w http.ResponseWriter, r *http.Request) {
 	component.Render(r.Context(), w)
 }
 
-func (app *Application) searchResults(w http.ResponseWriter, r *http.Request) {
-	component := components.SearchResults("Search", "Hepatitis")
+func (app *Application) formSearch(w http.ResponseWriter, r *http.Request) {
+	r.ParseForm()
+
+	searchTerm := r.Form["search"][0]
+	searchType := r.Form["options"][0]
+
+	search(w, r, app.repository, searchTerm, searchType, app.logger)
+}
+
+func search(w http.ResponseWriter, r *http.Request, rp *repository.Repository, searchTerm, searchType string, logger *slog.Logger) {
+	var result = &models.CodeSystemResultRow{}
+	defaultPageCount := 5
+
+	// retrieve code system
+	codeSystem, err := rp.GetCodeSystemsByLikeOID(r.Context(), searchTerm)
+	fmt.Println(codeSystem, err)
+	if err != nil || len(*codeSystem) < 1 {
+		if err == nil {
+			err = sql.ErrNoRows
+		}
+		handleError(err, searchTerm, logger, w, r)
```

Comment:
If I search something that doesn't exist (like `1.0.1`) I'll get a page that looks like this:
![Kapture 2024-09-25 at 10 05 57](https://github.com/user-attachments/assets/aacce19a-02c2-4b52-9bf4-d5b2e533d99e)

Do we need a `return` under this line so that only the error message is displayed? Right now we'll still get a table down below with 0 code system results.

There's also an additional `Error: Code System 1.0.1 not found` at the bottom of the page. I'm not immediately seeing how this is being rendered but I have a feeling you'll probably know where it's coming from 😃

Reply:
Ooooh good catch! I thought I fixed that earlier but I'll take another pass 👀